apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -1170,24 +1158,43 @@ def is_str_cat_col(field: pa.Field) -> bool:
             if name not in old_schema.names:
                 continue
             column = arrow_table.column(name)
-            old_info = old_schema.field(name)
-            new_info = new_schema.field(name)
-            if is_str_col(old_info) and is_str_cat_col(new_info):
-                # Convert from categorical-of-string to plain string.
-                column = column.to_pylist()
-            elif is_str_cat_col(old_info) and is_str_col(new_info):
-                # Convert from plain string to categorical-of-string.  Note:
+            old_field = old_schema.field(name)
+            new_field = new_schema.field(name)
+            if not is_cat(old_field) and is_cat(new_field):
+                # Convert from categorical to non-categorical.  Note that if
+                # this is a pa.ChunkedArray, there is no .dictionary_decode()
+                # for it, but each chunk does have a .dictionary_decode().
+
+                if isinstance(column, pa.ChunkedArray):
+                    column = pa.chunked_array(
+                        [chunk.dictionary_decode() for chunk in column.chunks]
+                    )
+                elif isinstance(column, pa.DictionaryArray):
+                    column = column.dictionary_decode()
+                else:
+                    column = column.to_pylist()
+
+            elif is_cat(old_field) and not is_cat(new_field):
+                # Convert from non-categorical to categorical.  Note:
                 # libtiledbsoma already merges the enum mappings, e.g if the
                 # storage has red, yellow, & green, but our new data has some
                 # yellow, green, and orange.
-                column = pa.array(
-                    column.to_pylist(),
-                    pa.dictionary(
-                        index_type=old_info.type.index_type,
-                        value_type=old_info.type.value_type,
-                        ordered=old_info.type.ordered,
-                    ),
-                )
+                if isinstance(column, pa.ChunkedArray):
+                    column = pa.chunked_array(
+                        [chunk.dictionary_encode() for chunk in column.chunks]
+                    )
+                elif isinstance(column, pa.DictionaryArray):
+                    column = column.dictionary_encode()
+                else:
+                    column = pa.array(
+                        column.to_pylist(),
```

Comment:
see comments above - under what cases does this get used?  (aka why does it need to be round-tripped through Python list?)